modules/7xbidBidAdapter.js

```diff
@@ -0,0 +1,157 @@
+import { registerBidder } from '../src/adapters/bidderFactory';
+
+const BIDDER_CODE = '7xbid';
+const BIDDER_ALIAS = '7xb';
+const ENDPOINT_BANNER = '//bidder.7xbid.com/api/v1/prebid/banner';
+const ENDPOINT_NATIVE = '//bidder.7xbid.com/api/v1/prebid/native';
+const COOKIE_SYNC_URL = '//bidder.7xbid.com/api/v1/cookie/gen';
+const SUPPORTED_MEDIA_TYPES = ['banner', 'native'];
+const SUPPORTED_CURRENCIES = ['USD', 'JPY'];
+const DEFAULT_CURRENCY = 'JPY';
+const NET_REVENUE = true;
+
+const _encodeURIComponent = function(a) {
+  let b = window.encodeURIComponent(a);
+  b = b.replace(/'/g, '%27');
+  return b;
+}
+
+export const _getUrlVars = function(url) {
+  var hash;
+  var myJson = {};
+  var hashes = url.slice(url.indexOf('?') + 1).split('&');
+  for (var i = 0; i < hashes.length; i++) {
+    hash = hashes[i].split('=');
+    myJson[hash[0]] = hash[1];
+  }
+  return myJson;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: [BIDDER_ALIAS], // short code
+  supportedMediaTypes: SUPPORTED_MEDIA_TYPES,
+  isBidRequestValid: function(bid) {
+    if (!(bid.params.placementId)) {
+      return false;
+    }
+
+    if (bid.params.hasOwnProperty('currency') &&
+      SUPPORTED_CURRENCIES.indexOf(bid.params.currency) === -1) {
+      utils.logInfo('Invalid currency type, we support only JPY and USD!')
+      return false;
+    }
+
+    return true;
+  },
+  /**
+    * Make a server request from the list of BidRequests.
+    *
+    * @param {validBidRequests[]} - an array of bids
+    * @return ServerRequest Info describing the request to the server.
+    */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    let serverRequests = [];
+    var refererInfo;
+    if (bidderRequest && bidderRequest.refererInfo) {
+      refererInfo = bidderRequest.refererInfo;
+    }
+    var g = (typeof (geparams) !== 'undefined' && typeof (geparams) == 'object' && geparams) ? geparams : {};
+    validBidRequests.forEach((bid, i) => {
+      let endpoint = ENDPOINT_BANNER
+      let data = {
+        'placementid': bid.params.placementId,
+        'cur': bid.params.hasOwnProperty('currency') ? bid.params.currency : DEFAULT_CURRENCY,
+        'ua': navigator.userAgent,
+        'adtk': _encodeURIComponent(g.lat ? '0' : '1'),
+        'loc': utils.getTopWindowUrl(),
+        'topframe': (window.parent === window.self) ? 1 : 0,
+        'sw': screen && screen.width,
+        'sh': screen && screen.height,
+        'cb': Math.floor(Math.random() * 99999999999),
+        'tpaf': 1,
+        'cks': 1,
+        'requestid': bid.bidId
+      };
+
+      if (bid.hasOwnProperty('nativeParams')) {
+        endpoint = ENDPOINT_NATIVE
+        data.tkf = 1 // return url tracker
+        data.ad_track = '1'
+        data.apiv = '1.1.0'
+      }
+
+      if (refererInfo && refererInfo.referer) {
+        data.referer = refererInfo.referer;
+      } else {
+        data.referer = '';
+      }
+
+      serverRequests.push({
```

Comment:
There was a push to have all adapters support SRA (Single Request Architecture).

I am unsure if this is till a requirement for new / existing adapters, so I have contacted the rest of the Prebid.org team and asked.

I will get back to you as soon as I get an answer, but wanted to give you a heads up!

Reply:
Ok got an answer and we will not require it.
But a future requirement may come up where we do, just for your information!

That change would have comms ahead of time and we would make it known to all adapters which need to be updated well ahead of time!